overlord/snapstate/handlers.go

```diff
@@ -778,8 +804,17 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.Active = false
 
-	pb := NewTaskProgressAdapterLocked(t)
-	err = m.backend.UnlinkSnap(oldInfo, pb)
+	// save the disabled services for later - note since this requires calling
+	// out to systemd, we need to unlock the state
+	st.Unlock()
```

Comment:
this unlocking here feels a bit problematic, it will make reasoning about this function harder over time, much better to unlock before most of the current code, query the services and have an explicit snapst.Last... = here